src/PoolManager.sol

```diff
@@ -207,34 +218,71 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC6909Claims {
         PoolId id = key.toId();
         _checkPoolInitialized(id);
 
-        key.hooks.beforeSwap(key, params, hookData);
-
-        uint256 feeForProtocol;
-        uint24 swapFee;
-        Pool.SwapState memory state;
-        (delta, feeForProtocol, swapFee, state) = pools[id].swap(
-            Pool.SwapParams({
-                tickSpacing: key.tickSpacing,
-                zeroForOne: params.zeroForOne,
-                amountSpecified: params.amountSpecified,
-                sqrtPriceLimitX96: params.sqrtPriceLimitX96
-            })
+        // The hook's deltas are from the point of view of the hook. Positive: the hook took money, negative: the hook sent money to the pool
+        // TODO consider adding hookDeltaInUnspecified to beforeSwap hook too
+        int128 hookDeltaInSpecified = key.hooks.beforeSwap(key, params, hookData);
+
+        bool exactInput = params.amountSpecified > 0;
+        params.amountSpecified -= hookDeltaInSpecified;
```

Comment:
I actually wonder for readability if we save this as a different variable like `swapAmount` because when reading params.amountSpecified below it seems like its the parameter when really the parameter has been changed. I had to double check that we were passing in the correct thing to swap.

Reply:
Ok we cant do this because of stack too deep. But I've grouped the lines differently, and added a comment to hopefully make it clear